megamek/src/megamek/client/ui/swing/MegaMekGUI.java

```diff
@@ -577,6 +599,86 @@ public String getDescription() {
                 hd.isRegister(), hd.isRegister() ? hd.getMetaserver() : "", null,
                 fc.getSelectedFile(), hd.getPlayerName());
     }
+
+    private boolean validateSaveVersion(final Node n) {
+        if (!n.hasChildNodes()) {
+            final String message = String.format(
+                    Messages.getString("MegaMek.LoadGameMissingVersion.message"),
+                    MMConstants.VERSION);
+            JOptionPane.showMessageDialog(frame, message,
+                    Messages.getString("MegaMek.LoadGameAlert.title"),
+                    JOptionPane.ERROR_MESSAGE);
+            LogManager.getLogger().error(message);
+        }
+        final NodeList nl = n.getChildNodes();
+        String release = null;
+        String major = null;
+        String minor = null;
+        String snapshot = null;
+        for (int i = 0; i < nl.getLength(); i++) {
+            final Node n2 = nl.item(i);
+            if (n2.getNodeType() != Node.ELEMENT_NODE) {
+                continue;
+            }
+
+            switch (n2.getNodeName()) {
+                case "release":
+                    release = n2.getTextContent();
+                    break;
+                case "major":
+                    major = n2.getTextContent();
+                    break;
+                case "minor":
+                    minor = n2.getTextContent();
+                    break;
+                case "snapshot":
+                    snapshot = n2.getTextContent();
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        final Version version = new Version(release, major, minor, snapshot);
+        if (MMConstants.VERSION.is(version)) {
+            return true;
+        } else {
+            final String message = String.format(
+                    Messages.getString("MegaMek.LoadGameIncorrectVersion.message"),
+                    version, MMConstants.VERSION);
+            JOptionPane.showMessageDialog(frame, message,
+                    Messages.getString("MegaMek.LoadGameAlert.title"), JOptionPane.ERROR_MESSAGE);
+            LogManager.getLogger().error(message);
+            return false;
+        }
+    }
+
+    private void parsePlayerNames(final Node n, final Vector<String> playerNames) {
+        if (!n.hasChildNodes()) {
+            return;
+        }
+
+        final NodeList nl = n.getChildNodes();
+        for (int i = 0; i < nl.getLength(); i++) {
+            final Node n2 = nl.item(i);
+            if ((n2.getNodeType() != Node.ELEMENT_NODE) || !n2.hasChildNodes()
+                    || !"megamek.common.Player".equals(n2.getNodeName())) {
```

Comment:
Would it make sense to use Player.class.getName() here? For the (unlikely) event of renaming or moving stuff

Reply:
Yes